gapis/api/vulkan/state_rebuilder.go

```diff
@@ -2778,18 +2778,20 @@ func (sb *stateBuilder) createQueryPool(qp QueryPoolObjectʳ) {
 	defer tsk.commit()
 	tsk.recordCmdBufCommand(func(commandBuffer VkCommandBuffer) {
 		for i := uint32(0); i < qp.QueryCount(); i++ {
-			if qp.Status().Get(i) != QueryStatus_QUERY_STATUS_INACTIVE {
-				sb.write(sb.cb.VkCmdBeginQuery(
-					commandBuffer,
-					qp.VulkanHandle(),
-					i,
-					VkQueryControlFlags(0)))
-			}
-			if qp.Status().Get(i) == QueryStatus_QUERY_STATUS_COMPLETE {
-				sb.write(sb.cb.VkCmdEndQuery(
-					commandBuffer,
-					qp.VulkanHandle(),
-					i))
+			switch qp.Status().Get(i) {
+			case QueryStatus_QUERY_STATUS_CREATED:
+				// do nothing
+			case QueryStatus_QUERY_STATUS_RESET:
+				sb.write(sb.cb.VkCmdResetQueryPool(commandBuffer, qp.VulkanHandle(), i, 1))
+			case QueryStatus_QUERY_STATUS_BEGAN:
+				sb.write(sb.cb.VkCmdBeginQuery(commandBuffer, qp.VulkanHandle(), i, VkQueryControlFlags(0)))
+			case QueryStatus_QUERY_STATUS_ENDED:
+				{
+					sb.write(sb.cb.VkCmdBeginQuery(commandBuffer, qp.VulkanHandle(), i, VkQueryControlFlags(0)))
+					sb.write(sb.cb.VkCmdEndQuery(commandBuffer, qp.VulkanHandle(), i))
```

Comment:
As mentioned above, there could be captures made with a gapid version after my change to vkCmdWriteTimestamp, and before this change, in which case we may have a timestamp query pool in this state, in which case begin/end is invalid. So I think we need to check whether this query pool is for timestamps, and treat this as TIMESTAMP_WRITTEN instead. Once we've added that check, the TIMESTAMP_WRITTEN might just be redundant.